packages/flame/lib/src/components/core/component.dart

```diff
@@ -611,16 +590,38 @@ class Component {
 
   /// Removes all the children for which the [test] function returns true.
   void removeWhere(bool Function(Component component) test) {
-    children.forEach((component) {
+    for (final component in children) {
```

Comment:
You don't know how happy this change makes me, my IDE always complains about this line 😂